### PR TITLE
ci: fix issue causing runner to crash in setup

### PR DESCRIPTION
**Pull request by clintonsteiner**
* caused by harden runner issues
    * set-output step also has been deprecated in github actions

Two failing pytest tests in latest here https://github.com/clintonsteiner/jenkinsapi/actions/runs/12392484959/job/34591759706

Solves this issue
https://github.com/pycontribs/jenkinsapi/issues/854

### CODE REVIEW

**Comment by clintonsteiner** [comment]
@Achimh3011 would you review - failures are rightful pytest failures that need addressed

**Comment by clintonsteiner** [comment]
https://github.com/clintonsteiner/jenkinsapi/tree/updateGithubActions
have this change next once this is merged

**Review by Achimh3011 — APPROVED** [review]

**Comment by clintonsteiner** [comment]
@Achimh3011 happy to fix several of these issues if you give me merge access. Would like to incrementally merge them in however

**Comment by clintonsteiner** [comment]
Hey @Achimh3011 will you take a look at https://github.com/pycontribs/jenkinsapi/pull/874

I merged all these sub prs into there for an easy merge

**Pull request closed without merging**
